vhdbuilder/packer/test/linux-vhd-content-test.sh

```diff
@@ -0,0 +1,261 @@
+#!/bin/bash
+
+testFilesDownloaded() {
+  test="testFilesDownloaded"
+  echo "$test:Start"
+  filesToDownload=$1
+
+  filesToDownload=$(echo $filesToDownload | jq -r ".[]" | jq . --monochrome-output --compact-output)
+  emptyFiles=()
+  missingPaths=()
+  for fileToDownload in ${filesToDownload[*]}; do
+    fileName=$(echo "${fileToDownload}" | jq .fileName -r)
+    downloadLocation=$(echo "${fileToDownload}" | jq .downloadLocation -r)
+    versions=$(echo "${fileToDownload}" | jq .versions -r | jq -r ".[]")
+
+    if [ ! -d $downloadLocation ]; then
+      err $test "Directory ${downloadLocation} does not exist"
+      missingPaths+=("$downloadLocation")
+      continue
+    fi
+
+    for version in ${versions}; do
+      file_Name=$(string_replace $fileName $version)
+      dest="$downloadLocation/${file_Name}"
+
+      if [ ! -s $dest ]; then
+        err $test "File ${dest} does not exist"
+        emptyFiles+=("$dest")
+        continue
+      fi
+    done
+
+    echo "---"
+  done
+  echo "$test:Finish"
+}
+
+testImagesPulled() {
+  test="testImagesPulled"
+  echo "$test:Start"
+  containerRuntime=$1
+  imagesToBePulled=$2
+  if [ $containerRuntime == 'containerd' ]; then
+    pulledImages=$(ctr -n k8s.io image ls)
+  elif [ $containerRuntime == 'docker' ]; then
+    pulledImages=$(docker images --format "{{.Repository}}:{{.Tag}}")
+  else
+    err $test "unsupported container runtime $containerRuntime"
+    return
+  fi
+
+  imagesNotPulled=()
+
+  imagesToBePulled=$(echo $imagesToBePulled | jq -r ".[]" | jq . --monochrome-output --compact-output)
+  for imageToBePulled in ${imagesToBePulled[*]}; do
+    downloadURL=$(echo "${imageToBePulled}" | jq .downloadURL -r)
+    versions=$(echo "${imageToBePulled}" | jq .versions -r | jq -r ".[]")
+
+    for version in ${versions}; do
+      download_URL=$(string_replace $downloadURL $version)
+
+      if [[ $pulledImages =~ $downloadURL ]]; then
+        echo "Image ${download_URL} has been pulled Successfully"
+      else
+        err $test "Image ${download_URL} has NOT been pulled"
+        imagesNotPulled+=("$download_URL")
+      fi
+    done
+
+    echo "---"
+  done
+  echo "$test:Finish"
+}
+
+err() {
+  echo "$1:Error: $2" >>/dev/stderr
+}
+
+string_replace() {
+  echo $1 | sed "s/\*/$2/"
+}
+
+filesToDownload='
```

Comment:
I will add these to a separate file in another PR which will include a Refactor for the install-dependencies.sh file